File: CSES PROBLEMS/INTRODUCTORY PROBLEMS/5. Permutations/main.cpp
#include <iostream> 
using namespace std;
int main() {
	int x; 
	cin >> x; 
	switch (x) {
		case 1: 
			cout << "1"; 
			break; 
		case 2: 
			cout << "NO SOLUTION"; 
			break; 
		case 3: 	
			cout << "NO SOLUTION"; 
			break; 
		default: 
			for (int i = 2; i <= x; i+=2){
				cout << i << " "; 		
			}
			for (int i = 1; i <= x; i+=2){
				cout << i << " "; 
			}
			break; 
	}
	return 0; 
}

File: CSES PROBLEMS/INTRODUCTORY PROBLEMS/5. Permutations/main.py
x = int(input())

if x == 1: 
    print(1)
    exit
if x < 4: 
    print("NO SOLUTION") 
else: 
    for a in range(2, x+1, 2): 
        print(str(a) + " ", end="") 
    for a in range(1, x+1, 2): 
        print(str(a) + " ", end="")
